src/views/Profile.vue

```diff
@@ -1,13 +1,369 @@
+          this.edit = false;
+
+          this.profile.first_name = this.user.first_name;
+          this.profile.last_name = this.user.last_name;
+
+          throw new Error(this.$t('notifyMessage.error.requiredNames'));
+        }
+
+        await userService.updateProfile(this.profile);
+        const profile = await userService.getProfile();
+
+        let tempUser = {};
+        tempUser.first_name = get(
+            profile,
+            'data.attributes.first_name',
+            this.user.first_name
+        );
+        tempUser.last_name = get(
+            profile,
+            'data.attributes.last_name',
+            this.user.last_name
+        );
+        tempUser.avatar = get(
+            profile,
+            'data.attributes.avatar',
+            'https://t4.ftcdn.net/jpg/03/46/93/61/360_F_346936114_RaxE6OQogebgAWTalE1myseY1Hbb5qPM.jpg'
+        );
+        this.setProfileState(tempUser);
+
+        this.$notify({
+          title: this.$t('notifyMessage.success.title'),
+          message: this.$t('notifyMessage.success.updateProfile'),
+          type: 'success',
+        });
+
+        this.edit = false;
+      } catch (error) {
+        this.notifyErrors(error);
+      }
+    },
+
+    async onClear() {
+      try {
+        await userService.clearAvatar();
+        const profile = await userService.getProfile();
+
+        let tempUser = {};
+        tempUser.first_name = get(
+            profile,
+            'data.attributes.first_name',
+            this.user.first_name
+        );
+        tempUser.last_name = get(
+            profile,
+            'data.attributes.last_name',
+            this.user.last_name
+        );
+        tempUser.avatar = get(
+            profile,
+            'data.attributes.avatar',
+            'https://t4.ftcdn.net/jpg/03/46/93/61/360_F_346936114_RaxE6OQogebgAWTalE1myseY1Hbb5qPM.jpg'
+        );
+        await this.setProfileState(tempUser);
+
+        this.edit = false;
+
+        this.$notify({
+          title: this.$t('notifyMessage.success.title'),
+          message: this.$t('notifyMessage.success.clearAvatar'),
+          type: 'success',
+        });
+      } catch (error) {
+        this.notifyErrors(error);
+      }
+    },
+
+    checkImage(image) {
+      const mimeTypes = ['image/jpg', 'image/jpeg', 'image/png', 'image/gif'];
+
+      try {
+        if(!mimeTypes.includes(image.target.files[0].type)) {
+          this.$refs.uploadAvatar.reset();
+          throw new Error(this.$t('notifyMessage.error.avatarMime'));
+        }
+
+        this.$refs.preview.classList.toggle("hidden");
+        this.$refs.avatarPreview.src = URL.createObjectURL(image.target.files[0]);
+        this.$refs.filenamePreview.innerHTML = (image.target.files[0].name);
+      } catch (error) {
+        this.notifyErrors(error);
+      }
+    },
+
+    async uploadAvatar() {
+      try {
+        const formData = new FormData();
+        formData.append('avatar', this.$refs.uploadAvatar[0].files[0]);
```

Comment:
Rather than using refs, we should save the `files` in a `data` variable instead.